tools/c7n_gcp/c7n_gcp/filters/metrics.py

```diff
@@ -0,0 +1,181 @@
+          aligner: ALIGN_COUNT
+          days: 14
+          value: 1
+          op: greater-than
+    """
+
+    schema = type_schema(
+        'metrics',
+        **{'name': {'type': 'string'},
+          'resource-key': {'type': 'string'},
+          'metric-key': {'type': 'string'},
+          'group-by-fields': {'type': 'array', 'items': {'type': 'string'}},
+          'days': {'type': 'number'},
+          'op': {'type': 'string', 'enum': list(OPERATORS.keys())},
+          'reducer': {'type': 'string', 'enum': REDUCERS},
+          'aligner': {'type': 'string', 'enum': ALIGNERS},
+          'value': {'type': 'number'},
+          'filter': {'type': 'string'},
+          'period': {'type': 'number'},
+          'missing-value': {'type': 'number'},
+          'required': ('value', 'name', 'op')})
+    permissions = ("monitoring.timeSeries.list",)
+
+    def process(self, resources, event=None):
+        days = self.data.get('days', 14)
+        duration = timedelta(days)
+
+        self.metric = self.data['name']
+        self.resource_key = self.data['resource-key']
+        self.metric_key = self.data['metric-key']
+        self.aligner = self.data.get('aligner', 'ALIGN_NONE')
+        self.reducer = self.data.get('reducer', 'REDUCE_NONE')
+        self.group_by_fields = self.data.get('group-by-fields', [])
+        self.missing_value = self.data.get('missing-value', 0)
+        self.end = datetime.now(pytz.timezone('UTC'))
+        self.start = self.end - duration
+        self.period = str((self.end - self.start).total_seconds()) + 's'
+        self.resource_metric_dict = {}
+        self.op = OPERATORS[self.data.get('op', 'less-than')]
+        self.value = self.data['value']
+        self.filter = self.data.get('filter')
+        self.c7n_metric_key = "%s.%s.%s" % (self.metric, self.aligner, self.reducer)
+
+        session = local_session(self.manager.session_factory)
+        client = session.client("monitoring", "v3", "projects.timeSeries")
+        project = session.get_default_project()
+
+        query_params = {
+            'filter': self.get_query_filter(resources),
+            'interval_startTime': self.start.isoformat(),
+            'interval_endTime': self.end.isoformat(),
+            'aggregation_alignmentPeriod': self.period,
+            "aggregation_perSeriesAligner": self.aligner,
+            "aggregation_crossSeriesReducer": self.reducer,
+            "aggregation_groupByFields": self.group_by_fields,
+            'view': 'FULL'
+        }
+        metric_list = client.execute_query('list', {'name': 'projects/' + project, **query_params})
+        if not metric_list.get('timeSeries'):
+            self.log.info("No metrics found for {}".format(self.c7n_metric_key))
+            return []
+
+        self.split_by_resource(metric_list['timeSeries'])
+        matched = [r for r in resources if self.process_resource(r)]
+
+        return matched
+
+    def get_query_filter(self, resources):
```

Comment:
this doesn't appear to be properly formatting the query wrt grouping of operator ie this should be

```
metric.type = "compute.googleapis.com/instance/cpu/usage_time"
AND (metric.label.instance_name = "instance-id" OR
  metric.label.instance_name = "other-instance-id")
```

Reply:
it was working without the `( )` , but for clarity and in case they change the syntax it makes sense to add them.

Reply:
cool, yeah that example above was from the docs, else its probably ambiguous.